### PR TITLE
Tomek/trustless platform script

### PR DESCRIPTION
**Pull request by banciur**

### CODE REVIEW

File: scripts/investIntoETO.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  console.log(`Token name: ${tokenName}, token symbol: ${tokenSymbol}`);
+
+  const etoState = (await eto.state()).toNumber();
+  console.log(`Eto is in ${CommitmentStateRev[etoState]} state`);
+  if (![CommitmentState.Public, CommitmentState.Whitelist].includes(etoState)) {
+    throw new Error(`Eto is in state that wont allow investment`);
+  }
+
+  // TODO: for nano we need instruction about how to setup derivation paths if user would like to
+  //  use non standard one
```
Comment (huesforalice) [inline scripts/investIntoETO.js:113]: I think this can be configured in the truffle config file, no?

File: scripts/investIntoETO.js [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+  // TODO: Think about checking if there was previous investment you can use investorTicket function and see what was there.
+  // Calculating contribution
+  const contribution = await eto.calculateContribution(account, false, contributionAmountToInvest);
+  const eligibleToInvest = contribution[1];
+  console.log(`Are you whitelisted: ${contribution[0]}`);
+  console.log(`Are you eligible to invest: ${eligibleToInvest}`);
+  console.log(`Minimum ticket: ${weiToEther(contribution[2])}`);
+  console.log(`Maximum ticket: ${weiToEther(contribution[3])}`);
+  console.log(`You will get ${contribution[4]} equity tokens`); // TODO: something is not right here in place where investorTicket is displayed we have to use fromWei - invesitgate
```
Comment (huesforalice) [inline scripts/investIntoETO.js:169]: What do you mean with this comment?

**Comment by banciur** [comment]
Add extra functionality to check if gas price in smart contracts is not outdated for case of ETH transfers. It might be useful for stage case.

File: scripts/investIntoETOCheck.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  const address = Accounts.prototype.privateKeyToAccount(options.pk).address;
+  console.log("Checking if investIntoETO script worked and investment has been made");
+  console.log(`Resolved PK to ${address}`);
+
+  const CONFIG = getConfig(web3, options.network, []);
+  const eto = await artifacts.require(CONFIG.artifacts.STANDARD_ETO_COMMITMENT).at(options.eto);
+  const ticket = await eto.investorTicket(address);
+  if (ticket[6].eq(0)) {
```
Comment (huesforalice) [inline scripts/investIntoETOCheck.js:32]: Is it not possible to check for the exact amount here?

**Review by huesforalice — DISMISSED** [review]

**Review by huesforalice — APPROVED** [review]

**Pull request merged**